samples/test/createAnalysis.test.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+'use strict';
+
+const {assert} = require('chai');
+const {afterEach, before, describe, it} = require('mocha');
+const cp = require('child_process');
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const {
+  ContactCenterInsightsClient,
+} = require('@google-cloud/contact-center-insights');
+const client = new ContactCenterInsightsClient();
+
+describe('CreateAnalysis', () => {
+  let projectId;
+  let conversationName;
+
+  before(async () => {
+    projectId = await client.getProjectId();
+  });
+
+  afterEach(() => {
+    client.deleteConversation({
```

Comment:
Does this return a promise?  Should we be awaiting?

Reply:
IIUC getting a promise is only used for async function. I made this a synchronous function instead of async because I want `deleteConversation` to run synchronously right after the test case. When I wrote this as an async function, the test case fails because there's no timing on when `deleteConversation` will run (e.g. sometimes it'll try to delete the resource before the test finishes creating the resource).

Reply:
I think I see how you got here.  The `client.deleteConversation` method is very much asynchronous.  If you don't `await` it, that doesn't make it synchronous - it just means the process can complete and exit before the operation is complete.  If the after hook is trying to delete the resource before it's created, that means we're missing an `await` somewhere during creation.  

I would suggest making the lambda passed into the `afterEach` method an `async` function, adding the `await` here, and extending the timeout (if needbe) in `samples/package.json`.  Otherwise, I'm worried the cleanup here isn't always going to complete before the process exits.

Reply:
Thanks for the suggestion! I updated 2 tests (CreateAnalysis and CreateConversation) to using `async after` but these are failing because of `NOT FOUND` errors. The other test (CreateConversationWithTtl) that's still using `synchronous after` is still passing. The error message from the 2 failing tests are `NOT FOUND`, saying there was no such resource to delete. But I queried these resources manually in my test project and found that they exist. I'm guessing I wrote `async after` wrong and thus the tests call `deleteConversation` before the conversation was created? Let me know if you're able to spot where I got it wrong 😣